### PR TITLE
[DEV-2498] Add PR deployment metrics

### PR DESCRIPTION
**Pull request by vmarkovtsev**

### CODE REVIEW

File: openapi.yaml [hunk 1]
```diff
@@ -3778,15 +3796,21 @@ components:
...
           description: Split by changed number of lines (inserted + removed).
           items:
             minimum: 0
             type: integer
           minLength: 2
           type: array
+        jira:
+          $ref: '#/components/schemas/JIRAFilter'
+        environments:
+          description: Calculate deployment metrics separately for each environment
```
Comment (warenlg) [inline openapi.yaml:3809]: Hence the combobox. Indeed, averaging the metrics for all envs does not make sense.

File: openapi.yaml [hunk 2]
```diff
@@ -3066,6 +3066,15 @@ components:
         * `pr-wait-first-review-time`: average time of waiting for the first review.
         * `pr-wait-first-review-count`: number of PRs used to calculate `pr-wait-first-review`. Note: this is *not* the same as the number of PRs waiting for the first review.
         * `pr-wait-first-review-count-q`: number of PRs used to calculate `pr-wait-first-review after applying the quantiles.
+        * `pr-deployment-time`: average time between when the PR was released and was deployed. We track deployments through `/events/deployments`. Requires setting `ForSet.environments`.
+        * `pr-deployment-count`: number of PRs used to calculate `pr-deployment-time` disregarding the quantiles. It equals to the overall number of deployed PRs in the given time window. Requires setting `ForSet.environments`.
+        * `pr-deployment-count-q`: number of PRs used to calculate `pr-deployment-time` after applying the quantiles. Requires setting `ForSet.environments`.
+        * `pr-lead-deployment-time`: extended Lead Time that includes the time passed since the PR was released until it was deployed. We discard PRs in repositories without deployments. Requires setting `ForSet.environments`.
```
Comment (warenlg) [inline openapi.yaml:3072]: Yes, this is great so that we keep pr-lead-time untouched.

**Review by warenlg — APPROVED** [review]
This is great!

File: openapi.yaml (hunk 1 above)
Comment (vmarkovtsev) [reply openapi.yaml:3809]: I remember that we discussed making this a setting near the release settings. But those settings do not satisfy the desire to quickly see how metrics change between several environments.

**Pull request merged**